torch/nn/functional.py

```diff
@@ -3391,7 +3391,7 @@ def kl_div(
     reduced = torch.kl_div(input, target, reduction_enum, log_target=log_target)
 
     if reduction == "batchmean" and input.dim() != 0:
-        reduced = reduced / input.size()[0]
+        reduced = reduced / torch.broadcast_shapes(input.size(), target.size())[0]
```

Comment:
curious, if `log_target` should also be taken into account here?

Reply:
Well, seems that `log_target` does not change the sizes of `target` and `input`? With my changes, I've tested that the results are the same when `log_target` is `True` or `False`.